### ISSUE TITLE
[CI] Failure in PreallocatedCircuitBreakerServiceTests.testRandom

### ISSUE THREAD

**Issue by dimitris-athanasiou** [elastic/elasticsearch#67256]
**Build scan**: https://gradle-enterprise.elastic.co/s/jv5etorzth6ma

**Repro line**: 

```
./gradlew ':server:test' --tests "org.elasticsearch.common.breaker.PreallocatedCircuitBreakerServiceTests.testRandom" -Dtests.seed=52ACD3BAE04922DC -Dtests.security.manager=true -Dtests.locale=sv -Dtests.timezone=Etc/GMT+0 -Druntime.java=8
```

**Reproduces locally?**: No

**Applicable branches**: 7.x

**Failure history**:

A couple more failures like this exist.

https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:PreallocatedCircuitBreakerServiceTests),sort:!(process.time-start,desc))

**Failure excerpt**:

```

org.elasticsearch.common.breaker.CircuitBreakingException: [parent] Data too large, data for [preallocate[test]] would be [489728634/467mb], which is larger than the limit of [489108275/466.4mb], real usage: [465253376/443.7mb], new bytes reserved: [24475258/23.3mb], usages [request=24475258/23.3mb, fielddata=0/0b, in_flight_requests=0/0b, accounting=0/0b] |  

at __randomizedtesting.SeedInfo.seed([52ACD3BAE04922DC:20E0F6B5512994AF]:0) |  
  |   | at org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService.checkParentLimit(HierarchyCircuitBreakerService.java:346) |  
  |   | at org.elasticsearch.common.breaker.ChildMemoryCircuitBreaker.addEstimateBytesAndMaybeBreak(ChildMemoryCircuitBreaker.java:109) |  
  |   | at org.elasticsearch.common.breaker.PreallocatedCircuitBreakerService.<init>(PreallocatedCircuitBreakerService.java:47) |  
  |   | at org.elasticsearch.common.breaker.PreallocatedCircuitBreakerServiceTests.preallocateRequest(PreallocatedCircuitBreakerServiceTests.java:144) |  
  |   | at org.elasticsearch.common.breaker.PreallocatedCircuitBreakerServiceTests.testRandom(PreallocatedCircuitBreakerServiceTests.java:96)

```

**Comment by dimitris-athanasiou** [comment]
Muted in `7.x`

**Comment by nik9000** [comment]
I tried to fix this with #67022. I imagine I didn't get it right.

**Comment by nik9000** [comment]
I believe #67022 fixed the failures that were happening on the 5th. The one on the 10th and 11th though - those are new! Debugging time!
